server/controllers/annotation_controller.js

```diff
@@ -5,67 +5,39 @@ import mongodb from 'mongodb';
 // direct access to a specific annotation
 export const getAnnotation = (user, annotationId) => {
   return Annotation.findById(annotationId)
-    .then(annotation => {
+    .then((annotation) => {
       if (annotation === null) {
         throw new Error('Annotation not found');
       }
 
-      let isAuthorized = annotation.isPublic;
       if (user !== null) {
-        isAuthorized = isAuthorized || user.isMemberOfAny(annotation.groups);
-      }
-
-      if (!isAuthorized) {
-        throw new Error('Not authorized to access this annotation');
+        const isAuthorized = annotation.isPublic || user.isMemberOfAny(annotation.groups);
+        if (!isAuthorized) {
+          throw new Error('Not authorized to access this annotation');
+        }
```

Comment:
Don't we want to keep checking that the annotation is public if user is null (not logged in)?